### PR TITLE
[REVIEW] Move search files to legacy

### PR DESCRIPTION
**Pull request by ChuckHastings**
Part 1 of the work to update search.hpp.

Addresses #2946 

Moves files to legacy directories in preparation for creating new experimental versions.

### CODE REVIEW

**Comment by jrhemstad** [comment]
This is targeted against `branch-0.10`. It should target `branch-0.11`, but when I try to update it I get the error: `There are no new commits between base branch 'branch-0.11' and head branch 'fea_port_search_to_new_column_part1'`

**Comment by ChuckHastings** [comment]
That's what happens when you hurry before running an errand.

Actually committed my changes.  Changed the merge target to branch-0.11.  Try reviewing this one :-)

**Comment by harrism** [comment]
@ChuckHastings Looks like you need to update some cython references to these files too.

**Comment by jrhemstad** [comment]
@ChuckHastings you need to update the `meta.yaml` file.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 - PR #2930 JSON Reader: Support ARROW_RANDOM_FILE input
 - PR #2987 Add `inplace` arg to `DataFrame.reset_index` and `Series`
+- PR #3025 Move current search code to legacy directories
```
Comment (harrism) [inline CHANGELOG.md]:
```suggestion
- PR #3025 Move search files to legacy
```

**Comment by jrhemstad** [comment]
@ChuckHastings please update with the latest branch-0.11 to unblock CI.

**Comment by ChuckHastings** [comment]
Still failing NVSTRING tests

**Comment by jrhemstad** [comment]
> Still failing NVSTRING tests

Yeah, this auto merged failed. https://github.com/rapidsai/cudf/pull/3089 It's being resolved.

**Comment by harrism** [comment]
Merge branch-0.11 into this one and fix conflicts and should be good to go.

**Pull request merged**